src/utils/get-package-name.js

```diff
@@ -17,10 +17,17 @@ const path = require('path');
 
 // This only works if tarball has been installed
 const getPackageName = (templateName) => {
-  if (templateName.match(/^\..+\.(tgz|tar(\.gz){0,1})$/)) {
+  if (templateName.match(/.+\.(tgz|tar(\.gz){0,1})$/)) {
+    const templatePath = path.resolve(templateName);
     const { dependencies } = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../../package.json')));
     return Object.entries(dependencies)
-      .find(([, version]) => version.match(templateName))[0];
+      .find(([, version]) => {
+        if (version.startsWith('file:')) {
+          const dependencyPath = path.resolve(__dirname, '../../', version.slice(5));
```

Comment:
curious, since it's hard to visualize, why `../../` and not `../` or `../../../`

Reply:
If I remember correctly, `__dirname` is the current javascript file's dir, so `../../` is reversing out `src/utils` to get back to the directory where the package.json is.  Then the path stored in the package.json with the "file:" prefix is a relative path from that directory.